enterprise/cmd/frontend/worker/auth/perms_syncer_cleaner.go

```diff
@@ -0,0 +1,110 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sourcegraph/sourcegraph/cmd/worker/job"
+	workerdb "github.com/sourcegraph/sourcegraph/cmd/worker/shared/init/db"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/internal/metrics"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+var _ job.Job = (*permissionsSyncerCleaner)(nil)
+
+// permissionsSyncerCleaner is a worker responsible for cleaning up processed
+// permission sync jobs.
+type permissionsSyncerCleaner struct{}
+
+func (p *permissionsSyncerCleaner) Description() string {
+	return "Removes completed or failed permissions sync jobs"
+}
+
+func (p *permissionsSyncerCleaner) Config() []env.Config {
+	return nil
+}
+
+func (p *permissionsSyncerCleaner) Routines(_ context.Context, observationCtx *observation.Context) ([]goroutine.BackgroundRoutine, error) {
+	db, err := workerdb.InitDB(observationCtx)
+	if err != nil {
+		return nil, errors.Wrap(err, "init DB")
+	}
+
+	m := metrics.NewREDMetrics(
+		observationCtx.Registerer,
+		"permission_sync_job_worker_cleaner",
+		metrics.WithCountHelp("Total number of permissions syncer cleaner executions."),
+	)
+	operation := observationCtx.Operation(observation.Op{
+		Name:    "PermissionsSyncer.Cleaner.Run",
+		Metrics: m,
+	})
+
+	return []goroutine.BackgroundRoutine{
+		goroutine.NewPeriodicGoroutineWithMetrics(
+			context.Background(),
+			"authz.permission_sync_job_worker_cleaner",
```

Comment:
I think it should match the worker name.

```suggestion
			"auth.permissions_syncer_cleaner",
```